File: src/main/java/br/com/blz/testjava/dto/ProductDto.java
package br.com.blz.testjava.dto;


public class ProductDto {

    private String sku;
    private String name;
    private boolean isMarketable;

    private InventoryDto inventory;


    public ProductDto(String sku, String name, InventoryDto inventory, boolean isMarketable) {
        super();
        this.sku = sku;
        this.name = name;
        this.inventory = inventory;
        this.isMarketable = isMarketable;
    }

    public ProductDto() {
        super();
    }

    public InventoryDto getInventory() {
        return inventory;
    }

    public void setInventory(InventoryDto inventory) {
        this.inventory = inventory;
    }

    public String getSku() {
        return sku;
    }

    public void setSku(String sku) {
        this.sku = sku;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isMarketable() {
        return isMarketable;
    }

    public void setMarketable(boolean isMarketable) {
        this.isMarketable = isMarketable;
    }
}

File: src/main/kotlin/br/com/blz/testjava/dto/ProductDto.kt
package br.com.blz.testjava.dto

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.EqualsAndHashCode
import org.hibernate.validator.constraints.Length
import javax.validation.constraints.NotEmpty

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
class ProductDto(
    @JsonProperty("sku")
    var sku: Long? = null,

    @JsonProperty("name")
    @get:NotEmpty(message = "Nome não pode ser vazio")
    @get:Length(min = 3, max = 200, message = "Nome do produto deve conter entre 3 e 200 caracteres")
    var name: String? = null,

    @JsonProperty("inventory")
    var inventory: InventoryDto? = null
) {
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    var isMarketable: Boolean? = false
}
